File: src/designer/libextension.cpp
#include <QAbstractExtensionFactory>
#include <QAbstractExtensionManager>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <extension.h>
#include "libextension.h"
#include "libextension.hxx"

QObject* QAbstractExtensionFactory_Extension(const QAbstractExtensionFactory* self, QObject* object, const libqt_string iid) {
    QString iid_QString = QString::fromUtf8(iid.data, iid.len);
    return self->extension(object, iid_QString);
}

void QAbstractExtensionFactory_OperatorAssign(QAbstractExtensionFactory* self, const QAbstractExtensionFactory* param1) {
    self->operator=(*param1);
}

void QAbstractExtensionFactory_Delete(QAbstractExtensionFactory* self) {
    delete self;
}

void QAbstractExtensionManager_RegisterExtensions(QAbstractExtensionManager* self, QAbstractExtensionFactory* factory, const libqt_string iid) {
    QString iid_QString = QString::fromUtf8(iid.data, iid.len);
    self->registerExtensions(factory, iid_QString);
}

void QAbstractExtensionManager_UnregisterExtensions(QAbstractExtensionManager* self, QAbstractExtensionFactory* factory, const libqt_string iid) {
    QString iid_QString = QString::fromUtf8(iid.data, iid.len);
    self->unregisterExtensions(factory, iid_QString);
}

QObject* QAbstractExtensionManager_Extension(const QAbstractExtensionManager* self, QObject* object, const libqt_string iid) {
    QString iid_QString = QString::fromUtf8(iid.data, iid.len);
    return self->extension(object, iid_QString);
}

void QAbstractExtensionManager_OperatorAssign(QAbstractExtensionManager* self, const QAbstractExtensionManager* param1) {
    self->operator=(*param1);
}

void QAbstractExtensionManager_Delete(QAbstractExtensionManager* self) {
    delete self;
}

File: src/designer/libextension.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionfactory.html)
pub const qabstractextensionfactory = struct {
    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionfactory.html#extension)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QtC.QAbstractExtensionFactory `
    ///
    /// ` object: QtC.QObject `
    ///
    /// ` iid: []const u8 `
    ///
    pub fn Extension(self: ?*anyopaque, object: ?*anyopaque, iid: []const u8) QtC.QObject {
        const iid_str = qtc.libqt_string{
            .len = iid.len,
            .data = iid.ptr,
        };
        return qtc.QAbstractExtensionFactory_Extension(@ptrCast(self), @ptrCast(object), iid_str);
    }

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionfactory.html#operator-eq)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QtC.QAbstractExtensionFactory `
    ///
    /// ` param1: QtC.QAbstractExtensionFactory `
    ///
    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QAbstractExtensionFactory_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionfactory.html#dtor.QAbstractExtensionFactory)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ## Parameter:
    ///
    /// ` self: QtC.QAbstractExtensionFactory `
    ///
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAbstractExtensionFactory_Delete(@ptrCast(self));
    }
};

/// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionmanager.html)
pub const qabstractextensionmanager = struct {
    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionmanager.html#registerExtensions)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QtC.QAbstractExtensionManager `
    ///
    /// ` factory: QtC.QAbstractExtensionFactory `
    ///
    /// ` iid: []const u8 `
    ///
    pub fn RegisterExtensions(self: ?*anyopaque, factory: ?*anyopaque, iid: []const u8) void {
        const iid_str = qtc.libqt_string{
            .len = iid.len,
            .data = iid.ptr,
        };
        qtc.QAbstractExtensionManager_RegisterExtensions(@ptrCast(self), @ptrCast(factory), iid_str);
    }

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionmanager.html#unregisterExtensions)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QtC.QAbstractExtensionManager `
    ///
    /// ` factory: QtC.QAbstractExtensionFactory `
    ///
    /// ` iid: []const u8 `
    ///
    pub fn UnregisterExtensions(self: ?*anyopaque, factory: ?*anyopaque, iid: []const u8) void {
        const iid_str = qtc.libqt_string{
            .len = iid.len,
            .data = iid.ptr,
        };
        qtc.QAbstractExtensionManager_UnregisterExtensions(@ptrCast(self), @ptrCast(factory), iid_str);
    }

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionmanager.html#extension)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QtC.QAbstractExtensionManager `
    ///
    /// ` object: QtC.QObject `
    ///
    /// ` iid: []const u8 `
    ///
    pub fn Extension(self: ?*anyopaque, object: ?*anyopaque, iid: []const u8) QtC.QObject {
        const iid_str = qtc.libqt_string{
            .len = iid.len,
            .data = iid.ptr,
        };
        return qtc.QAbstractExtensionManager_Extension(@ptrCast(self), @ptrCast(object), iid_str);
    }

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionmanager.html#operator-eq)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QtC.QAbstractExtensionManager `
    ///
    /// ` param1: QtC.QAbstractExtensionManager `
    ///
    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QAbstractExtensionManager_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qabstractextensionmanager.html#dtor.QAbstractExtensionManager)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ## Parameter:
    ///
    /// ` self: QtC.QAbstractExtensionManager `
    ///
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAbstractExtensionManager_Delete(@ptrCast(self));
    }
};
